tests/test_purge.py

```diff
@@ -190,3 +190,29 @@ def test_split_requests(client, requests_mocker):
     # The total set of objects sent to the API should be equal to that
     # provided to the client
     assert sorted(all_objects) == sorted(urls)
+
+
+@pytest.mark.xfail(
```

Comment:
Could this please be flipped around so it's a successful test, rather than an unsuccessful test marked with xfail?  In most of our projects we only use xfail to indicate "this is something we haven't got around to fixing yet".

When I suggested implementing a test to reproduce the bug, I meant a test that would fail, and then when you fix the bug you'd see the test starts to pass (i.e. test-driven development).  Not a test which passes on the broken code, then when you fix the bug it starts to fail and has to be marked as xfail.

As well as being potentially confusing, tests with xfail are weak tests, because they are counted as a "pass" if they fail for *any* reason - not only the reason you expected while writing them.  For instance if FastPurgeClient starts raising an unrelated exception which has nothing to do with auth or KeyError, this test won't be valid any more, but it'll still be considered a pass due to the xfail marker.